### PR TITLE
11 endpoint

### LINKED ISSUE #12: 11 Endpoint
11. Get Cash Dispensers Near a Market
Details:

1. The endpoint should be in the pattern of GET /api/v0/markets/:id/nearest_atms
2. You will need to utilize the [TomTom API](https://developer.tomtom.com/) for this. Specifically, the category search endpoint. Find a category that would work for ATM’s, and use the API to find ATM’s near the location of the Farmer’s Market.
3. The atms that are returned should be in the order of closest to furthest away.
4. If an invalid market id is passed in, a 404 status as well as a descriptive error message should be sent back in the response.
5. The data top level key should always point to an array even if one or zero atms were located near the market location.

Example #1 😁
Request:

   GET /api/v0/markets/327794/nearest_atms
   Content-Type: application/json
   Accept: application/json
Response: status: 200

 {
     "data": [
         {
             "id": null,
             "type": "atm",
             "attributes": {
                 "name": "ATM",
                 "address": "3902 Central Avenue Southeast, Albuquerque, NM 87108",
                 "lat": 35.07904,
                 "lon": -106.60068,
                 "distance": 0.10521432030421865
             }
         },
         {
             "id": null,
             "type": "atm",
             "attributes": {
                 "name": "ATM",
                 "address": "4100 Central Avenue Southeast, Albuquerque, NM 87108",
                 "lat": 35.0788,
                 "lon": -106.59842,
                 "distance": 0.14448001321588486
             }
         },
         ...,
         ...,
         ...,
     ]
 }

Example #2 😭
Request:

     GET /api/v0/markets/123123123123/nearest_atm (where `123123123123` is an invalid Market id)
     Content-Type: application/json
     Accept: application/json
Response: status: 404

 {
     "errors": [
         {
             "detail": "Couldn't find Market with 'id'=123123123123"
         }
     ]
 }

### PR DESCRIPTION
**Pull request by yainporter**
## Type of change
- [x] New feature
- [ ] Bug Fix
- [ ] Test
- [ ] Refactor/Style

## Description
Relevant User Story(s): 
End Point 11
*Brief summary of the change(s) included in this pull request*
- Added Faraday
- Added Webmock
- Added VCR
- Generate API key
- Create Atm Poro
- Create Atm Service
- Create Atm Facade
- Create NearestAtm route
- Create NearestAtm Controller

description closes #12 

## Check the correct boxes
- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes
- [ ] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed

*Please describe what in the world happened if tests are broken*

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

### CODE REVIEW

File: spec/facades/atm_facade_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    describe "#sorted_atms" do
+      it "sorts the atms by closest to farthest", :vcr do
+        sorted_atms = @atm_facade.sorted_atms
+
+        consecutive_sorted = sorted_atms.each_cons(2).all? do |first, second|
+          first.distance <= second.distance
+        end
+
+        expect(consecutive_sorted).to eq(true)
+      end
```
Comment (yainporter) [inline spec/facades/atm_facade_spec.rb:45]: The #each_cons method iterates over a pair in the array, and checks that the block statement is true for all pairs

File: app/controllers/api/v0/market_vendors_controller.rb [hunk 2]
```diff
@@ -1,17 +1,10 @@
...
+      render json: no_market_or_vendor, status: :bad_request
     else
-      market = Market.find(params[:market_id])
-      vendor = Vendor.find(params[:vendor_id])
-      # These two methods help us raise the RecordNotFound, 404 response we want for ids passed that don't exist
-      market.vendors << vendor
-      # This is another way to create MarketVendor, since we want to check for the RecordNotFound, this will also
-      # raise the 422 error we want for a record that already exists
-      render json: MarketVendorSerializer.new(MarketVendor.find_by(market_vendor_params)).serialize_success, status: :created
+      render json: successful_creation, status: :created
```
Comment (odellmac4) [inline app/controllers/api/v0/market_vendors_controller.rb:7]: Nice refactoring here

File: app/facades/atm_facade.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  def create_atms
+    list_of_atms.map do |atm|
+      Atm.new(atm)
+    end
+  end
+
+  def sorted_atms
+    create_atms.sort_by{ |atm| atm.distance }
+  end
+end
```
Comment (odellmac4) [inline app/facades/atm_facade.rb:21]: nice and clean!! great work here

File: spec/facades/atm_facade_spec.rb (hunk 1 above)
Comment (odellmac4) [reply spec/facades/atm_facade_spec.rb:45]: nice! this is bringing back memories from mod1 when I had to use this for battleship

**Review by odellmac4 — APPROVED** [review]

**Pull request merged**